src/adapter/aws-lambda/handler.ts

```diff
@@ -564,7 +564,7 @@ const isProxyEventV2 = (event: LambdaEvent): event is APIGatewayProxyEventV2 =>
  * @returns True if the content type is binary, false otherwise.
  */
 export const defaultIsContentTypeBinary = (contentType: string): boolean => {
-  return !/^(text\/(plain|html|css|javascript|csv).*|application\/(.*json|.*xml).*|image\/svg\+xml.*)$/.test(
+  return !/^(text\/(plain|html|css|javascript|csv).*|application\/(.*json$|.*json\.|.*xml$|.*xml\.).*|image\/svg\+xml.*)$/.test(
```

Comment:
Thank you for considering the changes.

As indicated in #4470, I believe we should make the following changes, including improvements to the current regular expression.

```suggestion
  return !/^text\/(?:plain|html|css|javascript|csv)|(?:\/|\+)(?:json|xml)(?:;|$)/test(
```